File: collect_app/src/main/java/org/odk/collect/android/formentry/RefreshFormListDialogFragment.java
package org.odk.collect.android.formentry;

import android.content.Context;

import androidx.annotation.NonNull;

import org.odk.collect.material.MaterialProgressDialogFragment;

public class RefreshFormListDialogFragment extends MaterialProgressDialogFragment {

    protected RefreshFormListDialogFragmentListener listener;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        if (context instanceof RefreshFormListDialogFragmentListener) {
            listener = (RefreshFormListDialogFragmentListener) context;
        }
        setTitle(getString(org.odk.collect.strings.R.string.downloading_data));
        setMessage(getString(org.odk.collect.strings.R.string.please_wait));
        setCancelable(false);
    }

    @Override
    protected String getCancelButtonText() {
        return getString(org.odk.collect.strings.R.string.cancel_loading_form);
    }

    @Override
    protected OnCancelCallback getOnCancelCallback() {
        return () -> {
            listener.onCancelFormLoading();
            dismiss();
            return true;
        };
    }

    public interface RefreshFormListDialogFragmentListener {
            void onCancelFormLoading();
    }
}

File: collect_app/src/main/java/org/odk/collect/android/formentry/RefreshFormListDialogFragment.kt
package org.odk.collect.android.formentry

import android.content.Context
import android.content.DialogInterface
import org.odk.collect.android.R
import org.odk.collect.material.MaterialProgressDialogFragmentNew

class RefreshFormListDialogFragment : MaterialProgressDialogFragmentNew() {
    lateinit var listener: RefreshFormListDialogFragmentListener

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is RefreshFormListDialogFragmentListener) {
            listener = context
        }
        title = getString(R.string.downloading_data)
        message = getString(R.string.please_wait)
        canBeCanceled = false
        negativeButtonTitle = getString(R.string.cancel_loading_form)
        negativeButtonListener =
            DialogInterface.OnClickListener { _: DialogInterface?, _: Int ->
                listener.onCancelFormLoading()
                dismiss()
            }
    }

    interface RefreshFormListDialogFragmentListener {
        fun onCancelFormLoading()
    }
}
